part_2/second_split/method_v2.R

```diff
@@ -0,0 +1,398 @@
+rm(list = ls())
+setwd("/Volumes/GoogleDrive/My Drive/alsta_analysis_LA/second_split")
+
+######################################
+##### Section 1: Import packages #####
+######################################
+library(forecast)
+library(bsts)
+library(glmnet)
+library(plyr)
+library(lme4)
+library(strucchange)
+library(changepoint)
+library(mice)
+source("functions_v2.R")
+
+####################################
+##### Section 2: Preprocessing #####
+####################################
+
+### Impute data ###
+# Remove future data.
+end_date <- as.POSIXlt("2018-06-21", tz = "America/Los_Angeles")
+
+# Impute range: 2017-08-01 to 2017-09-01.
+# Read the original data to get the memory information.
+data_agg_sum <- readRDS("data_agg_sum.rds")
+data_impute_prep_mem <- data_agg_sum[data_agg_sum$date < end_date,]
+data_impute_prep_mem[data_impute_prep_mem$memory_gib_seconds > 5e4,] # 2018-05-01
+data_agg_sum[as.character(data_agg_sum$date) == "2018-05-01", "memory_gib_seconds"] <- mean(data_agg_sum[as.character(data_agg_sum$date) == "2018-04-30", "memory_gib_seconds"],
+                                                                                            data_agg_sum[as.character(data_agg_sum$date) == "2018-05-02", "memory_gib_seconds"])
+
+data_impute_prep_mem <- data_agg_sum[data_agg_sum$date < end_date,]
+data_impute_prep_mem <- data_impute_prep_mem[data_impute_prep_mem$date <= as.POSIXct("2017-09-02", tz = "America/Los_Angeles") & 
+                                               data_impute_prep_mem$date >= as.POSIXct("2017-08-01", tz = "America/Los_Angeles"),
+                                             ]
+data_impute_prep <- data_agg_sum[data_agg_sum$date < end_date,]
+data_impute_prep <- data_impute_prep[data_impute_prep$date > as.POSIXct("2017-09-02", tz = "America/Los_Angeles") | 
+                                       data_impute_prep$date < as.POSIXct("2017-08-01", tz = "America/Los_Angeles"),]
+data_impute_prep <- data_impute_prep[data_impute_prep$Month %in% month.abb[c(5:9)], ]
+summary(data_impute_prep)
+head(data_impute_prep)
+
+# Impute using Mixed effect model.
+fit_impute <- lm(gcu_seconds ~ memory_gib_seconds + Year + Month + Week + Day,
```

Comment:
The comment says that you will be using a mixed effects model. Also, I'm not sure you will want to incorporate Day as a predictor (unless you are using some sort of tree-based method that can also act as a feature reduction).